src/server/plugins/internal/firestore.ts

```diff
@@ -89,6 +100,48 @@ const listenToTransactions = (
     })
 }
 
+/**
+ * Listens to events that happen in the "consents" collection.
+ * Note that when the server starts running, all existing documents in Firestore
+ * will be treated as a document that is created for the first time. The handler for
+ * `onCreate` consent must be able to differentiate whether a document is created in
+ * realtime or because it has persisted in the database when the server starts.
+ *
+ * @param server a server object as defined in the hapi library.
+ * @param options a configuration object for the plugin.
+ * @returns a function to unsubscribe the listener.
+ */
+const listenToConsents = (
+  server: StateServer,
+  options: Options
+): (() => void) => {
+  const consentHandlers = options.handlers.consents
+
+  return firebase
+    .firestore()
+    .collection('consents')
+    .onSnapshot((querySnapshot) => {
+      querySnapshot.docChanges().forEach((change) => {
+        if (change.type === 'added' && consentHandlers.onCreate) {
```

Comment:
minor nit: are there no enums/constants to compare change type - why are comparing change.type to a hardcoded string

Reply:
I see your point - so I copied this portion from stevent's implementation of the `transactions` listener and I think steven copied this directly from the firebase docs
no harm in being more struct here with an enum, but given the limited scope - not sure its necessary

Reply:
leaving this here for @eoln 's thoughts